### PR TITLE
[NBS 1.0] Create migration guide to stop support the legacy backend system in plugins

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Update the plugin migration guide with steps to stop supporting the legacy backend system.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/backend-system/building-plugins-and-modules/08-migrating.md [hunk 1]
```diff
@@ -255,3 +255,67 @@ backend.add(
 Checkout the [custom service implementations](https://backstage.io/docs/backend-system/building-backends/index#custom-service-implementations) documentation and also the [core service configurations](https://backstage.io/docs/backend-system/core-services/index) page in case you'd like to create your own custom mock factory for one or more services.
 
 3. Now you can finally start your plugin locally by running `yarn start` from the root folder of your plugin.
+
+## Remove support for the old backend system
```
Comment (Rugvip) [inline docs/backend-system/building-plugins-and-modules/08-migrating.md:259]: I think it's worth getting guidance for plugin libraries in there straight away too. Main thing there being to remove all dependencies on `backend-common`

**Review by Rugvip — COMMENTED** [review]
Nice! 👍

File: docs/backend-system/building-plugins-and-modules/08-migrating.md (hunk 1 above)
Comment (camilaibs) [reply docs/backend-system/building-plugins-and-modules/08-migrating.md:259]: Updated here, is it good to go?
https://github.com/backstage/backstage/pull/26133/commits/e4fdf1f02d5fcfb82a9c687b5ed97264693646d6

Comment (Rugvip) [reply docs/backend-system/building-plugins-and-modules/08-migrating.md:259]: It's not super explicit, but it does cover it 😅. Let's go with this, I imagine the fact that those packages and all exports are deprecated will likely be enough of a nudge in the end too.

**Review by Rugvip — APPROVED** [review]
:shipit: 😎

**Pull request merged**